File: utilities/nonvisualstudio/elixir/Spec/Seq.Spec.ex
# IN POWERSHELL
# iex.bat - REPL
# elixir c:\Users\Thomas\Documents\GitHub\practice\utilities\nonvisualstudio\elixir\Spec\Seq.Spec.ex

Code.require_file "c:/Users/Thomas/Documents/GitHub/practice/utilities/nonvisualstudio/elixir/Implementation/Seq.ex", __DIR__

ExUnit.start

defmodule SeqTests do
  use ExUnit.Case, async: true

  test "pmap(1..10, fn (elem) -> (elem * elem) end) should return Enum.map(1..10, fn(elem) -> elem * elem) end." do
    assert Seq.pmap(1..10, fn (elem) -> (elem * elem) end) == Enum.map(1..10, fn(elem) -> elem * elem end)
  end

  test "pmap(1..100, fn (elem) -> (elem * elem) end) should return Enum.map(1..100, fn(elem) -> elem * elem) end." do
    assert Seq.pmap(1..100, fn (elem) -> (elem * elem) end) == Enum.map(1..100, fn(elem) -> elem * elem end)
  end

end

File: utilities/nonvisualstudio/haskell/Spec/Seq.Spec.hs
{-
ghci c:\Users\Thomas\Documents\GitHub\practice\utilities\nonvisualstudio\haskell\Spec\Seq.Spec.hs
     c:\Users\Thomas\Documents\GitHub\practice\utilities\nonvisualstudio\haskell\Implementation\Seq.hs
-}
-- :r - reload
-- :q - quit
-- :set +s for times

module SeqTests where

  import Test.HUnit
  import System.IO

  import Seq

  powModTests = TestList
    [
      TestCase $ assertEqual "pMap negate 1..10000 should return map negate 1..10000." (map negate [1..10000]) (pMap negate [1..10000])
    ]

  testCases = TestList
    [
      powModTests
    ]

  tests = runTestTT testCases
